File: Assets/_scripts/Player.js
class Player extends MonoBehaviour
{

    public var acceleration : float = 100;
    public var firingPower: float = 1000;
    public var gun : Transform;
    public var bullet : GameObject;
    
    public var speed : float = 4; 
    
    // function Start(){
    //     Fire();
    // }
    
        
    function Awake() {
        Screen.SetResolution(480, 320, true);
    }
    
    function FixedUpdate(){
       rigidbody.velocity = transform.forward * speed;
    }
    
    function OnApplicationQuit(){
        Profiler.PrintResults();
    }

    function Update () {

        var rotation = transform.localEulerAngles;
        rotation += 2 * Vector3(-Input.GetAxis ("Mouse Y"), Input.GetAxis ("Mouse X"), 0);
        transform.localEulerAngles = rotation;

        if(Input.GetKey("w")){
            rigidbody.AddForce(transform.forward * acceleration);
        }
        if(Input.GetKey("s")){
            rigidbody.AddForce(transform.forward * -acceleration);
        }

        if(Input.GetKey("a")){        
            rigidbody.AddForce(transform.right * -acceleration);
        }
        if(Input.GetKey("d")){        
            rigidbody.AddForce(transform.right * acceleration);
        }

        fireIfNeeded();
    }

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


    function fireIfNeeded(){
       if(Input.GetKeyDown(KeyCode.Space)) {
           Fire();
       }
    }
    
    function Fire(){
        var newBullet : GameObject = Instantiate(bullet, gun.position, transform.rotation);
        Physics.IgnoreCollision(collider, newBullet.collider);
        newBullet.transform.localScale = bullet.transform.lossyScale;
        newBullet.rigidbody.velocity = rigidbody.velocity;
        var spear : Spear = newBullet.GetComponent(Spear);
        spear.FireWithForce(firingPower);        
    }
}

File: Assets/_scripts/Player.cs
using UnityEngine;
using System.Collections;

public class Player : MonoBehaviour
{
    public Transform gun;
    public GameObject bullet ;
    public float firingPower = 2000;

    public float speed  = 4; 

    // void Start(){
    //     Fire();
    // }


    void Awake() {
        Screen.SetResolution(480, 320, true);
    }

    void FixedUpdate(){
       rigidbody.velocity = transform.forward * speed;
    }

    void OnApplicationQuit(){
        Profiler.PrintResults();
    }

    void Update () {

        Vector3 rotation = transform.localEulerAngles;
        rotation += 2 * new Vector3(-Input.GetAxis ("Mouse Y"), Input.GetAxis ("Mouse X"), 0);
        transform.localEulerAngles = rotation;

        fireIfNeeded();
    }

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


    void fireIfNeeded(){
       if(Input.GetKeyDown(KeyCode.Space)) {
           Fire();
       }
    }

    void Fire(){
        GameObject newBullet  = (GameObject)Instantiate(bullet, gun.position, transform.rotation);
        Physics.IgnoreCollision(collider, newBullet.collider);
        newBullet.transform.localScale = bullet.transform.lossyScale;
        newBullet.rigidbody.velocity = rigidbody.velocity;
        Spear spear  = (Spear)newBullet.GetComponent(typeof(Spear));
        spear.FireWithForce(firingPower);        
    }
}